### PR TITLE
Add logic to retry only for limited list of recoverable URLErrors

### PR DESCRIPTION
**Pull request by addb**
## Description

Screenshot 2024-05-15 at 1 18 33 PM

- Added logic to retry hits failed with a recoverable URLError, which were previously discarded.
- Added tests.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -25,5 +25,5 @@ analyzer_rules:
 - unused_import
 
 function_body_length:
-    warning: 50
-    error: 150
\ No newline at end of file
+    warning: 60
```
[the remark this answers is not in the document]
Comment (kevinlind) [reply .swiftlint.yml:28]: One suggestion is to take lines 132-148 in the AnalyticsHitProcessor and put into a separate function `getResponseEventData` (or something like that).

**Review by kevinlind — COMMENTED** [review]
Looks good. Minor suggestion to keep the functions smaller.

File: AEPAnalytics/Sources/AnalyticsHitProcessor.swift [hunk 2]
```diff
@@ -164,14 +165,24 @@ class AnalyticsHitProcessor: HitProcessing {
...
-        } else if connection.responseCode == nil {
-            // retry this hit later if connection response code is nil (no network connectivity)
-            Log.warning(label: LOG_TAG, "\(#function) - Retrying Analytics hit, there is currently no network connectivity")
-            completion(false)
         } else {
-            // unrecoverable error. delete the hit from the database and continue
-            Log.warning(label: LOG_TAG, "\(#function) - Dropping Analytics hit, request with url \(url.absoluteString) failed with error \(connection.error?.localizedDescription ?? "") and unrecoverable status code \(connection.responseCode ?? -1)")
-            completion(true)
+            // handle network transport error
+            if let urlError = connection.error as? URLError, urlError.isRecoverable {
```
Comment (kevinlind) [inline AEPAnalytics/Sources/AnalyticsHitProcessor.swift:170]: Previously we would retry when `connection.responseCode == nil`, which from my understanding would be from any connection error. Now, however, we will only retry from a small list of URLErrors. Is there a concern that after this change some previously recoverable errors will no longer be recoverable?

Comment (addb) [reply AEPAnalytics/Sources/AnalyticsHitProcessor.swift:170]: Yes we covered this in the guild meeting and we will follow the same pattern of retrying only for limited set of errors in all our extensions.

**Review by kevinlind — APPROVED** [review]

**Pull request merged**